test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.Protocol.cs

```diff
@@ -379,6 +379,39 @@ public class Protocol
                 }
             }
 
+            [Fact]
+            public async Task HandlerIsRemovedProperlyWithOff()
+            {
+                var connection = new TestConnection();
+                var hubConnection = CreateHubConnection(connection);
+                var handlerCalled = new TaskCompletionSource<int>();
+                try
+                {
+                    await hubConnection.StartAsync().OrTimeout();
+
+                    hubConnection.On<int>("Foo", (val) =>
+                    {
+                        handlerCalled.TrySetResult(val);
+                    });
+
+                    hubConnection.Off("Foo");
+                    await connection.ReceiveJsonMessage(new { invocationId = "1", type = 1, target = "Foo", arguments = 1 }).OrTimeout();
+                    var handlerTask = handlerCalled.Task;
+
+                    // Give the handler task 4 seconds to complete. It shouldn't because we've removed it. 
+                    var completedTask = await Task.WhenAny(Task.Delay(4000), handlerTask).OrTimeout();
```

Comment:
4 seconds is a long delay for a unit test. Is this waiting on an internal timeout?

Is there a smarter way to test this? Or can the timeout be reduced?

Reply:
4 seconds is just shorter than the `OrTimeout()` period. The timeout can be reduced

Reply:
OrTimeout has an overload to customize the timeout period